apps/db/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/db/InntektsmeldingRepository.kt

```diff
@@ -137,14 +113,22 @@ class InntektsmeldingRepository(
 
     fun hentNyesteInntektsmeldingSkjema(forespoerselId: UUID): SkjemaInntektsmelding? =
         transaction(db) {
-            hentNyesteImSkjemaQuery(forespoerselId)
+            InntektsmeldingEntitet
+                .selectAll()
+                .where { InntektsmeldingEntitet.forespoerselId eq forespoerselId }
+                .orderBy(InntektsmeldingEntitet.innsendt, SortOrder.DESC)
+                .limit(1)
                 .firstOrNull()
                 ?.getOrNull(InntektsmeldingEntitet.skjema)
         }
 
-    fun hentNyesteBerikedeInntektsmeldingId(forespoerselId: UUID): UUID? =
+    fun hentNyesteInntektsmeldingId(forespoerselId: UUID): UUID? =
         transaction(db) {
-            hentNyesteImQuery(forespoerselId)
+            InntektsmeldingEntitet
+                .selectAll()
+                .where { (InntektsmeldingEntitet.forespoerselId eq forespoerselId) and InntektsmeldingEntitet.inntektsmeldingId.isNotNull() }
```

Comment:
Her har jeg endret `InntektsmeldingEntitet.dokument.isNotNull()` til `InntektsmeldingEntitet.inntektsmeldingId.isNotNull()`. En forskjell er at `inntektsmeldingId` lagres tidligere, som resulterer i en liten endring i bruken i `LagreJournalpostIdRiver.kt`. Kort sagt så strammer vi inn for når vi sender en IM videre til Spleis når vi har to IM-er for samme sak i systemet samtidig.

Reply:
Grunnen til at vi brukte `InntektsmeldingEntitet.dokument.isNotNull()` er fordi at `inntektsmeldingId` før ble lagret samtidig som `dokument`, men det ble endret for lenge siden.